rust/cardano-blockchain-types/src/network.rs

```diff
@@ -22,29 +21,46 @@ pub(crate) const ENVVAR_MITHRIL_EXE_NAME: &str = "MITHRIL_EXE_NAME";
 
 /// Enum of possible Cardano networks.
 #[derive(
-    Debug,
-    Copy,
-    Clone,
-    PartialEq,
-    Eq,
-    Hash,
-    PartialOrd,
-    Ord,
-    strum::EnumIter,
-    strum::VariantNames,
-    strum::EnumString,
-    strum::Display,
+    Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, strum::VariantNames, strum::Display,
 )]
-#[strum(ascii_case_insensitive)]
 #[strum(serialize_all = "snake_case")]
-#[repr(usize)]
+#[non_exhaustive]
 pub enum Network {
     /// Cardano mainnet network.
-    Mainnet = 0,
+    Mainnet,
     /// Cardano pre-production network.
-    Preprod = 1,
+    Preprod,
     /// Cardano preview network.
-    Preview = 2,
+    Preview,
+    /// Cardano devnet network.
+    Devnet {
+        /// Mithril signature genesis key for a blockchain.
+        genesis_key: &'static str,
+        /// Cardano blockchain networking magic number genesis block setting
+        magic: u64,
+        /// Cardano blockchain network id genesis block setting
+        network_id: u64,
+        /// Cardano byron epoch length genesis block setting
+        byron_epoch_length: u32,
+        /// Cardano byron slot length genesis block setting
+        byron_slot_length: u32,
+        /// Cardano byron known slot genesis block setting
+        byron_known_slot: u64,
+        /// Cardano byron known hash genesis block setting
+        byron_known_hash: &'static str,
+        /// Cardano byron known time genesis block setting
+        byron_known_time: u64,
+        /// Cardano shelley epoch length genesis block setting
+        shelley_epoch_length: u32,
+        /// Cardano shelley slot length genesis block setting
+        shelley_slot_length: u32,
+        /// Cardano shelley known slot genesis block setting
+        shelley_known_slot: u64,
+        /// Cardano shelley known hash genesis block setting
+        shelley_known_hash: &'static str,
+        /// Cardano shelley known time genesis block setting
+        shelley_known_time: u64,
+    },
```

Comment:
I'm surprised that the [large_enum_variant](https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant) lint isn't triggered there. 🙃